File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        if target < nums[l]:
            return 0
        elif target > nums[r]:
            return r + 1

        while r >= l:
            mid = l + (r - l) // 2

            if nums[mid] == target:
                return mid
            elif target > nums[mid]:
                l = mid + 1
            elif target < nums[mid]:
                r = mid - 1

        return l

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0, r = numbers.length - 1;

        while(r > l){
            int left = numbers[l];
            int right = numbers[r];
            int sum = left + right;

            if(sum == target) return new int[]{l + 1, r + 1};

            if(sum > target) r--;
            else l++;
        }

        return null;
    }
}
